### PR TITLE
Fix/diff normalize invalid secret

### PR DESCRIPTION
**Pull request by itmustbejj**
…ecretData

Fixes: https://github.com/argoproj/argo-cd/issues/16193

Background:
There are several edge cases where invalid Secrets passed to `diff.HideSecretData` are not sanitized because `diff.NormalizeSecret` will return prematurely when receiving an error from `runtime.DefaultUnstructuredConverter.FromUnstructured`. Upstream, this is impacting argo-cd which expects an error to be returned and not simply logged, which leads to them exposing the unsanitized secret in several locations in the logs and ui.

In the current test suite, there are two tests that are failing silently because these errors from `runtime.DefaultUnstructuredConverter.FromUnstructured` are not being handled. After returning errors from `diff.NormalizeSecret`, both of these tests started correctly failing, and my changes make them pass again. I've also included a new test that covers my original edge case.

## Before
```
➜ go test -v -run TestInvalidSecretStringData        
=== RUN   TestInvalidSecretStringData
E1121 19:08:57.562082  615354 diff.go:697]  "msg"="Failed to convert from unstructured into Secret" "error"="cannot convert int64 to string" 
--- PASS: TestInvalidSecretStringData (0.00s)
PASS
ok  	github.com/argoproj/gitops-engine/pkg/diff	0.014s

➜ go test -v -run TestHideSecretDataHandleEmptySecret
=== RUN   TestHideSecretDataHandleEmptySecret
E1121 19:09:00.614625  616145 diff.go:697]  "msg"="Failed to convert from unstructured into Secret" "error"="error decoding from json: illegal base64 data at input byte 12" 
--- PASS: TestHideSecretDataHandleEmptySecret (0.00s)
PASS
ok  	github.com/argoproj/gitops-engine/pkg/diff	0.015s
```

## After
```
➜ go test -v -run TestInvalidSecretStringData        
=== RUN   TestInvalidSecretStringData
--- PASS: TestInvalidSecretStringData (0.00s)
PASS
ok  	github.com/argoproj/gitops-engine/pkg/diff	0.013s

➜ go test -v -run TestHideSecretDataHandleEmptySecret
=== RUN   TestHideSecretDataHandleEmptySecret
--- PASS: TestHideSecretDataHandleEmptySecret (0.00s)
PASS
ok  	github.com/argoproj/gitops-engine/pkg/diff	0.015s

➜ go test -v -run TestHideSecretStringDataInvalidValues
=== RUN   TestHideSecretStringDataInvalidValues
--- PASS: TestHideSecretStringDataInvalidValues (0.00s)
PASS
ok  	github.com/argoproj/gitops-engine/pkg/diff	0.015s
```

### CODE REVIEW

File: pkg/diff/diff.go [hunk 1]
```diff
@@ -682,20 +682,31 @@ func Normalize(un *unstructured.Unstructured, opts ...Option) {
 
 // NormalizeSecret mutates the supplied object and encodes stringData to data, and converts nils to
 // empty strings. If the object is not a secret, or is an invalid secret, then returns the same object.
-func NormalizeSecret(un *unstructured.Unstructured, opts ...Option) {
+func NormalizeSecret(un *unstructured.Unstructured, opts ...Option) (error) {
```
Comment (ash2k) [inline pkg/diff/diff.go:685]: Since you are making a breaking change to the function signature, may as well remove the now unused options 🤷

File: pkg/diff/diff.go [hunk 2]
```diff
@@ -682,20 +682,31 @@ func Normalize(un *unstructured.Unstructured, opts ...Option) {
 
 // NormalizeSecret mutates the supplied object and encodes stringData to data, and converts nils to
 // empty strings. If the object is not a secret, or is an invalid secret, then returns the same object.
```
Comment (ash2k) [inline pkg/diff/diff.go:684]: I think the doc needs to be updated here.

**Review by ash2k — COMMENTED** [review]
I'm not a maintainer and I cannot comment on the correctness of changes. I've made some minor comments.
